### DISCUSSION TITLE
[Question]  Sorry if this has been asked already. I looked at the Wiki on how to split up my store into separate slices, but how can I go about this using Typescript?

### DISCUSSION THREAD

**Discussion by sagar-monga** [pmndrs/zustand#2078] [Q&A]
I tried to create 2 slices, let's say User & Item.

```
export type UserSlice = {
  name: string;
  age: number;
  email: string;

  setName: (name: string) => void;
  setAge: (age: string) => void;
  setEmail: (email: string) => void;
}

export type ItemSlice = {
  name: string;
  qty: number;

  setName: (name: string) => void;
  setQty: (qty: number) => void;
}
```

Then I created the slices using `StateCreator` with the types mentioned above.

The problem arises here, I want to combine the states such that the RootState is
```
export type RootState = {
  user: UserSlice,
  items: ItemSlice[],
}
```

But when I use the this in `create<RootState>`, I face an error stating the type mismatch.

What would be the best approach for handling this?

**Comment by dbritto-dev** [comment]
@sagar-monga ts versions requires curried version `create<RootState>()(...)` but js version can use both curried version and uncurried version `create(...)` if you want to us ts you should always use curried version

**Reply by sagar-monga** [reply]
I am using the ts version, so curried. But my object state is different in both cases as I mentioned. Looking for a solution for that

**Reply by dbritto-dev** [reply]
> I am using the ts version, so curried. But my object state is different in both cases as I mentioned. Looking for a solution for that
> 

@sagar-monga you shouldn't face any issues. Please share us a ts playground demo so we can help you
